packages/x-date-pickers/src/tests/describeGregorianAdapter/testCalculations.ts

```diff
@@ -164,122 +277,343 @@ export const testCalculations: DescribeGregorianAdapterTestSuite = ({ adapter })
+    });
+
+    it('should work with different timezones', function test() {
+      if (!adapter.isTimezoneCompatible) {
+        this.skip();
+      }
 
-    const nextYearLocale = adapter.addYears(testDateLocale, -1);
-    expect(adapter.isBeforeYear(nextYearLocale, testDateLocale)).to.equal(true);
-    expect(adapter.isBeforeYear(testDateLocale, nextYearLocale)).to.equal(false);
+      // Both dates below have the same timestamp, but they are not in the same year when represented in their respective timezone.
+      // The adapter should still consider that they are in the same year.
+      const londonTimezone = adapterTZ.endOfYear(
+        adapterTZ.setTimezone(testDateIso, 'Europe/London'),
+      );
+      const parisTimezone = adapterTZ.setTimezone(londonTimezone, 'Europe/Paris');
+
+      expect(adapterTZ.isBeforeYear(londonTimezone, parisTimezone)).to.equal(false);
+      expect(adapterTZ.isBeforeYear(parisTimezone, londonTimezone)).to.equal(false);
+    });
   });
 
-  it('Method: isBeforeDay', () => {
-    const nextDayIso = adapter.addDays(testDateIso, -1);
-    expect(adapter.isBeforeDay(nextDayIso, testDateIso)).to.equal(true);
-    expect(adapter.isBeforeDay(testDateIso, nextDayIso)).to.equal(false);
+  describe('Method: isBeforeDay', () => {
+    it('should work with the same timezone', () => {
+      const previousDayIso = adapter.addDays(testDateIso, -1);
+      expect(adapter.isBeforeDay(previousDayIso, testDateIso)).to.equal(true);
+      expect(adapter.isBeforeDay(testDateIso, previousDayIso)).to.equal(false);
+
+      const previousDayLocale = adapter.addDays(testDateLocale, -1);
+      expect(adapter.isBeforeDay(previousDayLocale, testDateLocale)).to.equal(true);
+      expect(adapter.isBeforeDay(testDateLocale, previousDayLocale)).to.equal(false);
+    });
+
+    it('should work with different timezones', function test() {
+      if (!adapter.isTimezoneCompatible) {
+        this.skip();
+      }
+
+      // Both dates below have the same timestamp, but they are not in the same day when represented in their respective timezone.
+      // The adapter should still consider that they are in the same day.
+      const londonTimezone = adapterTZ.endOfDay(
+        adapterTZ.setTimezone(testDateIso, 'Europe/London'),
+      );
+      const parisTimezone = adapterTZ.setTimezone(londonTimezone, 'Europe/Paris');
+
+      expect(adapterTZ.isBeforeDay(londonTimezone, parisTimezone)).to.equal(false);
+      expect(adapterTZ.isBeforeDay(parisTimezone, londonTimezone)).to.equal(false);
 
-    const nextDayLocale = adapter.addDays(testDateLocale, -1);
-    expect(adapter.isBeforeDay(nextDayLocale, testDateLocale)).to.equal(true);
-    expect(adapter.isBeforeDay(testDateLocale, nextDayLocale)).to.equal(false);
+      // Both dates below have the same day when represented in their respective timezone,
+      // But not when represented in the same timezone
+      // The adapter should consider that they are not in the same day
```

Comment:
Same question regarding the correctness of this comment, because there is a manipulation of hours happening. 🤔